dali/ft/filecopy.cpp

```diff
@@ -2563,7 +2573,8 @@ void FileSprayer::setSource(IFileDescriptor * source, unsigned copy, unsigned mi
                 FilePartInfo & next = * new FilePartInfo(rfn);
                 Owned<IPartDescriptor> part = source->getPart(idx);
                 next.extractExtra(*part);
-                next.size = multi.getSize(i);
+                if (copy)
+                    next.size = multi.getSize(i);
```

Comment:
Not sure I understand this change.. is it related to this JIRA?

Reply:
Yes, it is related and I will explain.

Reply:
So, the reason for that check was: don't set the file part size here - force to check disk.
Later I realised there were a couple, redundant gatherFileSizes(false); function calls in the setSource() methods (like this). I removed them, based on the spray() always call that function with errorIfMissing = true.